tests/modules/configs/test_config.py

```diff
@@ -121,3 +120,134 @@ def test_get_conf_path_file_not_found(self):
             conf.get_conf_path(conf_path)
         assert ex.type == FileNotFoundError
         assert err_msg.get_conf_path.get("file_not_found") in str(ex.value)
+
+
+class TestMergedConf:
+    """Class to test functions of MergedConf Class"""
+
+    @staticmethod
+    def none_to_dict(obj):
+        if obj is None:
+            return {}
+        return obj
+
+    @pytest.mark.parametrize("dct, expect", [
+        [merge_conf.dct1, merge_conf.expect1],
+        [merge_conf.dct2, merge_conf.expect2],
+        [merge_conf.dct3, merge_conf.expect3],
+    ])
+    def test_merge_conf(self, dct, expect):
+        conf = MergedConf(Loggers({"debug": True}))
+        assert conf.merge_conf(dct) == expect
+
+    @pytest.mark.parametrize("merged_key, rule_type", [
+        ["resources_thread_count", "exclusive_over_defined"]
+    ])
+    def test_apply_config_rule_exclusive_over_defined(self, merged_key, rule_type):
```

Comment:
We should really check for conflicts between configs not internally. If we are expecting duplicates within each config, first of all we should handle that in the source code (perhaps using jsonschema or https://github.com/dmeranda/demjson).